plexapi/library.py

```diff
@@ -436,21 +441,80 @@ def reload(self, key=None):
         return self._server.library.section(self.title)
 
     def edit(self, agent=None, **kwargs):
-        """ Edit a library (Note: agent is required). See :class:`~plexapi.library.Library` for example usage.
+        """ Edit a library. See :class:`~plexapi.library.Library` for example usage.
 
             Parameters:
+                agent (str, optional): The library agent.
                 kwargs (dict): Dict of settings to edit.
+
+            Example:
+
+                .. code-block:: python
+                LibrarySection.edit(**{'location': ['/path/1', 'path/2', '/path/3']})
         """
         if not agent:
             agent = self.agent
-        part = '/library/sections/%s?agent=%s&%s' % (self.key, agent, urlencode(kwargs))
+
+        if kwargs['location']:
+            paths = kwargs.get('location')
+            for path in paths:
+                if not utils.pathExist(path):
+                    raise BadRequest('Path: %s does not exist.' % path)
+        params = list(kwargs.items())
+
+        part = '/library/sections/%s?agent=%s&%s' % (self.key, agent, urlencode(params, doseq=True))
         self._server.query(part, method=self._server._session.put)
 
         # Reload this way since the self.key dont have a full path, but is simply a id.
         for s in self._server.library.sections():
             if s.key == self.key:
                 return s
 
+    def addLocations(self, location):
+        """ Add a location to a library.
+
+            Parameters:
+                location (str, list, or class): A single folder path, list of paths,
+                    or :class:`~plexapi.library.Location`.
+
+            Example:
+
+                .. code-block:: python
+                LibrarySection.addLocations('/path/1')
+                LibrarySection.addLocations(['/path/1', 'path/2', '/path/3'])
+                MovieLibrarySection.addLocations(ShowLibrarySection._locations()[0])
```

Comment:
Avoid private methods in docs.

`MovieLibrarySection.addLocations(ShowLibrarySection.locations[0])`

Same for `removeLocations()`.

Reply:
My example is showing how to use the `Location` class. In your example `ShowLibrarySection.locations[0]` returns the path as `str`. 

It could be that we don't need `Location` classes to used.

Reply:
`Location` objects are not exposed to the user anywhere so it's unlikely they would come across it. If they do come across it, the class is listed as an accepted parameter in the docstring. Most users would use a location string since that is more natural.